assets/js/collaborative-editor/components/AIAssistantPanel.tsx

```diff
@@ -0,0 +1,601 @@
+      ? 'Ask me anything about this job...'
+      : sessionType === 'workflow_template'
+        ? 'Ask me anything about this workflow...'
+        : 'Ask me anything...';
+
+  const disabledMessage =
+    view === 'sessions' && (!hasSessionContext || !hasCompletedSessionLoad)
+      ? 'Loading conversations...'
+      : view === 'chat' && connectionState !== 'connected'
+        ? 'Connecting...'
+        : undefined;
+
+  // Load session list when viewing sessions
+  useEffect(() => {
+    if (!isOpen || view !== 'sessions' || !storeSessionType) return;
+
+    if (!hasSessionContext) {
+      return;
+    }
+
+    void loadSessionList();
+  }, [
+    isOpen,
+    view,
+    storeSessionType,
+    jobCodeContext?.job_id,
+    hasSessionContext,
+    loadSessionList,
+  ]);
+
+  // Re-fetch session list when tab becomes visible (handles browser tab sleep)
+  useEffect(() => {
+    if (!isOpen || view !== 'sessions' || !hasSessionContext) return;
+
+    const handleVisibilityChange = () => {
+      if (document.visibilityState === 'visible') {
+        void loadSessionList();
+      }
+    };
+
+    document.addEventListener('visibilitychange', handleVisibilityChange);
+    return () => {
+      document.removeEventListener('visibilitychange', handleVisibilityChange);
+    };
+  }, [isOpen, view, hasSessionContext, loadSessionList]);
+
+  const handleShowSessions = () => {
+    setView('sessions');
+    setIsMenuOpen(false);
+
+    if (onShowSessions) {
+      onShowSessions();
+    }
+  };
+
+  const handleToggleMenu = () => {
+    setIsMenuOpen(prev => !prev);
+  };
+
+  const handleOpenAbout = () => {
+    setIsAboutOpen(true);
+    setIsMenuOpen(false);
+  };
+
+  const handleSessionSelect = (selectedSessionId: string) => {
+    if (onSessionSelect) {
+      onSessionSelect(selectedSessionId);
+      setView('chat');
+    }
+  };
+
+  const handleClose = () => {
+    if (sessionId) {
+      if (onShowSessions) {
+        onShowSessions();
+      }
+    } else {
+      onClose();
+    }
+  };
+
+  useEffect(() => {
+    if (!isMenuOpen) return;
+
+    const handleClickOutside = (event: MouseEvent) => {
+      const target = event.target as HTMLElement;
+      if (
+        !target.closest('[data-menu-trigger]') &&
+        !target.closest('[data-menu-content]')
+      ) {
+        setIsMenuOpen(false);
+      }
+    };
+
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => document.removeEventListener('mousedown', handleClickOutside);
+  }, [isMenuOpen]);
+
+  return (
+    <div
+      data-testid="ai-assistant-panel"
+      data-session-type={sessionType || undefined}
+      className={cn(
+        'h-full flex flex-col overflow-hidden bg-white relative',
+        !isResizable && [
+          'flex-none border-l border-gray-200',
+          'transition-[width,border] duration-300 ease-in-out',
+          isOpen ? 'w-[400px]' : 'w-0 border-l-0',
+        ]
+      )}
+      role="dialog"
+      aria-modal="false"
```

Comment:
This isn't a modal, we're setting it to "dialog" and then turning off focus trapping with `aria-modal="false"`, essentially undoing the point of a dialog.

The main panel should use role="complementary" (or just be an `<aside>`) since it's a persistent side panel that coexists with main content.

And there are two internal overlays that might need to be `aria-modal`s. The about modal and the disclaimer screen, not deal breakers since their focus trapping impact is limited. 

The way to test the UX for this is by pressing enter or tab after you open and close things, it shouldn't be tabbing to different inputs behind things you can't see (this is what `role=dialog` and `aria-modal` essentially controls)

Reply:
Changed the main panel to use a semantic <aside> element which has an implicit role="complementary" - much better suited for a persistent side panel that doesn't trap focus. Also added proper role="dialog" aria-modal="true" to the internal overlays (About modal and Disclaimer screen) that actually do behave as modal dialogs with trapped focus.